bootloader/piboot.go

```diff
@@ -375,7 +383,55 @@ func (p *piboot) layoutKernelAssetsToDir(snapf snap.Container, dstDir string) er
 	return nil
 }
 
+func (p *piboot) eepromVersionSupportsTryboot() bool {
+	// To find out the EEPROM version we do the same as the
+	// rpi-eeprom-update script (see
+	// https://github.com/raspberrypi/rpi-eeprom/blob/master/rpi-eeprom-update)
+	buf, err := ioutil.ReadFile(rpi4EepromTimeStampPath)
+	if err != nil {
+		return false
+	}
+
+	// The timestamp is seconds since the epoch, UTC time
+	eepromTs := binary.BigEndian.Uint32(buf)
+	// 2021-03-18 or more modern supports tryboot, see
+	// https://github.com/raspberrypi/rpi-eeprom/blob/master/firmware/release-notes.md#2021-04-19---promote-2021-03-18-from-latest-to-default---default
+	// The timestamp we compare with (jue 18 mar 2021 08:54:11 UTC) can be found with:
+	// $ strings pieeprom-2021-03-18.bin | grep BUILD_TIMESTAMP
+	if eepromTs < 1616057651 {
```

Comment:
Should we do something like:
```
t := time.Date(2021, time.June, 18, 8, 54, 11, 0, time.UTC)
if eepromTs < t.Unix() {
```

instead so it's more self-documenting (and ensures correctness)? Having said that, I'm getting 1624006451 here.

Reply:
I see, fair enough, I thought the timestamp from grep is actually in the human-readable format. In that case it makes sense to have it in seconds.